### DISCUSSION TITLE
Variables Node

### DISCUSSION THREAD

**Discussion by brainlessbitch** [YaLTeR/niri#418] [Ideas]
Allowing a node for custom variables would be really nice, something like

```kdl
variables {
  terminal "alacritty"
}

binds {
  Mod+T { spawn variables.terminal; }
}
```
not sure if this is possible, I know very little about kdl

**Comment by YaLTeR** [comment]
I don't think this would be valid KDL syntax, and in KDL2 this would be a string. We could maybe come up with some special KDL syntax for this (using a type name comes to mind like `(var)terminal`).

The problem is, it will blow up the config parsing code in size in complexity, since suddenly every argument needs to be able to be a variable, which need to be expanded after the first parsing pass. So I'm quite reluctant.

**Comment by brainlessbitch** [comment]
that's fair, my friend actually wrote a sexpression version of kdl that supports variables, so I might just end up finding a way to execute that then evaluate the kdl output
